pkg/tracker/reward/reward.go

```diff
@@ -0,0 +1,412 @@
+		case event := <-events:
+			err := self.recordGasUsage(event)
+			if err != nil {
+				level.Error(self.logger).Log("msg", "record gas usage", "err", err)
+			}
+			err = self.recordGasUsageEstimated(self.ctx)
+			if err != nil {
+				level.Error(self.logger).Log("msg", "record gas usage estimation", "err", err)
+			}
+		}
+	}
+}
+
+func (self *RewardTracker) recordGasUsage(event *tellor.TellorNonceSubmitted) error {
+	receipt, err := self.client.TransactionReceipt(self.ctx, event.Raw.TxHash)
+	if err != nil {
+		return errors.Wrap(err, "receipt retrieval")
+	} else if receipt != nil && receipt.Status == types.ReceiptStatusSuccessful { // Failed transactions cost is monitored in a different process.
+		// tx, _, err := self.client.TransactionByHash(self.ctx, event.Raw.TxHash)
+		// if err != nil {
+		// 	level.Error(self.logger).Log("msg", "get transaction by hash", "err", err)
+		// 	return
+		// }
+		level.Debug(self.logger).Log("msg", "adding gas used", "gasUsed", receipt.GasUsed)
+		if err := self.addGasUsed(event.Slot.String(), receipt.GasUsed); err != nil {
+			return errors.Wrap(err, "adding gas used")
+		}
+		return nil
+	}
+	return errors.New("transaction not yet mined")
+}
+
+func (self *RewardTracker) addGasUsed(slot string, gasUsed uint64) error {
+	var err error
+	appender := self.tsDB.Appender(self.ctx)
+
+	// Round up the time so that all appends happen with the same TS and
+	// avoid out of order samples errors.
+	ts := timestamp.FromTime(time.Now().Round(5 * time.Second))
+
+	defer func() { // An appender always needs to be committed or rolled back.
+		if err != nil {
+			if err := appender.Rollback(); err != nil {
+				level.Error(self.logger).Log("msg", "db rollback failed", "err", err)
+			}
+			return
+		}
+		if errC := appender.Commit(); errC != nil {
+			err = errors.Wrap(err, "db append commit failed")
+		}
+	}()
+	lbls := labels.Labels{
+		labels.Label{Name: "__name__", Value: "gas_usage_actual"},
+		labels.Label{Name: "slot", Value: slot},
+	}
+
+	_, err = appender.Append(0, lbls, ts, float64(gasUsed))
+	if err != nil {
+		return errors.Wrap(err, "append values to the DB")
+	}
+	return nil
+}
+
+func (self *RewardTracker) recordGasUsageEstimated(ctx context.Context) error {
+	ctx, cncl := context.WithTimeout(ctx, 2*time.Second)
+	defer cncl()
+	// Getting abi.
+	abi, err := abi.JSON(strings.NewReader(tellor.TellorABI))
+	if err != nil {
+		return errors.Wrap(err, "getting abi")
+	}
+
+	// Getting current slot.
+	slot, err := self.Slot()
+	if err != nil {
+		return errors.Wrap(err, "getting current slot")
+	}
+
+	// Getting current challenge.
+	vars, err := self.contractInstance.GetNewCurrentVariables(&bind.CallOpts{Context: ctx})
+	if err != nil {
+		return errors.Wrap(err, "call GetNewCurrentVariables")
+	}
+	// We don't have actual values here, so we'll use math.MaxInt64 as our values.
+	// As we tested, this won't make any significant difference in the estimated gas value!
```

Comment:
I prefer if we don't use `we`
```suggestion
	// Use arbitrary values as when tested didn't make a noticeable difference between using real vs arbitrary values.
```